### DISCUSSION TITLE
Gpu

### DISCUSSION THREAD

**Discussion by brrbrrry** [brkirch/stable-diffusion-webui#20] [Ideas]
Do you think it will ever be possible to use the arm gpu with sd or is there any forks that could lead me in the right direction or would it be less efficient either way heard something about ml. I'm rambling but I'm trying to get faster generations if anyone can point me in the right direction.

**Comment by zihang-huang** [comment]
Apple has officially released a coreML implementation of stable diffusion that utilizes GPUs and the acceleration units on the Apple Silicon. 

[ml-stable-diffusion](https://github.com/apple/ml-stable-diffusion)

[swift-coreml-diffusers](https://github.com/huggingface/swift-coreml-diffusers)

However, I don't think sdwebui will ever use these libraries, as they requires different models. Also its functionality are also very limited compares to what sdwebui can do right now.

**Comment by jrittvo** [comment]
I'd love to read an official answer on this issue, but I am under the impression that M1/M2 Macs are using the GPU cores for at least some parts of the Automatic1111 pipeline through the MPS (Metal Performance Shaders) backend called by Python.  Empirically, the system monitoring apps I have show that, during image generation, my GPU utilization rises to over 90% and GPU memory load (not sure what that means in the context of Mac's unified memory system) sits at about 50%.  I think all of this is impacted by the specific arguments that you run with.  So while no CoreML is correct, that isn't the only way to access compute units beside the CPU cores.
